app/[...slug]/page.tsx

```diff
@@ -1,9 +1,21 @@
+import { Metadata } from 'next';
 import { getStoryblokApi, ISbStoriesParams, StoryblokClient } from '@storyblok/react/rsc';
 import StoryblokStory from '@storyblok/react/story';
 import { resolveRelations } from '@/utilities/resolveRelations';
+import { getPageMetaData } from '@/utilities/getPageMetadata';
 
 const activeEnv = process.env.NODE_ENV || 'development';
 
+export async function generateMetadata({ params }): Promise<Metadata> {
+  const { data } = await getStoryData(params);
+  const blok = data.story.content;
+
+  let slug: string = params.slug ? params.slug.join('/') : '';
+  const meta = getPageMetaData({ blok, slug });
```

Comment:
We already know the slug from params that were generated. Passing that into the getPageMetadata function to create the self-referencing canonical URLs for each page (when there's no custom canonical URL entered on Storyblok)